tests/jupyterhub_test.py

```diff
@@ -16,11 +16,33 @@ def setup_module(module):
 
 
 def test_check_auth_jupyterhub(fc):
+    import sirepo.srdb
+    from pykern import pkio
+    from pykern.pkunit import pkeq
+
+    def _get_num_jupyterhub_user_dirs():
+        return len(pkio.sorted_glob(sirepo.srdb.root().join("jupyterhub/user/**"))) - 1
+
+    def _jupyter_user_dir_exists(uid):
+        for x in [
```

Comment:
I think this could be simpler:

`sirepo.srdb.root().join("jupyterhub", "user", uid.lower()).exists(dir=True)`

Reply:
sometimes the jupyter username is prefixed with guest-, so this won't work

Reply:
We should know when this happens, and we should code for it. At a minimum, this test should work:

`bool(len(pkio.sorted_glob(sirepo.srdb.root().join("jupyterhub", "user", f"*{uid.lower()}*")))`

This is a cleaner condition, because it leverages glob as opposed to working around glob. The `**` in the code is not right, because it shouldn't be recursive. The user directories are at a flat level. uid is random and that's a good enough test on a clean db, which this is.